src/script/lua_api/l_vmanip.cpp

```diff
@@ -280,16 +285,16 @@ int LuaVoxelManip::l_get_param2_data(lua_State *L)
 	bool use_buffer  = lua_istable(L, 2);
 
 	MMVManip *vm = o->vm;
-
-	u32 volume = vm->m_area.getVolume();
+	const u32 volume = vm->m_area.getVolume();
 
 	if (use_buffer)
 		lua_pushvalue(L, 2);
 	else
 		lua_createtable(L, volume, 0);
 
 	for (u32 i = 0; i != volume; i++) {
-		lua_Integer param2 = vm->m_data[i].param2;
+		// Do not push unintialized data to Lua
+		lua_Integer param2 = (vm->m_flags[i] & VOXELFLAG_NO_DATA) ? 0 : vm->m_data[i].getParam2();
```

Comment:
`vm->m_flags` and `vm->m_data` could be moved outside the loop if the compiler is not smart enough to do it automatically.

The same remark for `LuaVoxelManip::l_get_data`.

Reply:
I hope it is.